### PR TITLE
Clean up grdmath usage

### PR DESCRIPTION
**Pull request by PaulWessel**
Part of the work in #5341 . Also fixes an issue in the **-I** usage.

### CODE REVIEW

File: src/grdmath.c [hunk 1]
```diff
@@ -167,293 +167,296 @@ static void Free_Ctrl (struct GMT_CTRL *GMT, struct GRDMATH_CTRL *C) {	/* Deallo
...
-	GMT_Message (API, GMT_TIME_NONE, "usage: %s [%s]\n\t[%s]\n\t[-D<resolution>][+f] [%s]\n\t[-M] [-N] [-S] [%s] [%s] [%s] [%s]\n\t[%s] [%s]"
-		" [%s]\n\t[%s] [%s]\n\t[%s] [%s] [%s]\n\t%s [%s]", name, GMT_Rgeo_OPT, GMT_A_OPT, GMT_I_OPT, GMT_V_OPT, GMT_a_OPT, GMT_bi_OPT, GMT_di_OPT,
+	GMT_Usage (API, 0, "usage: %s [%s] [%s] [-D<resolution>[+f] [%s] [-M] [-N] [-S] [%s] [%s] [%s] [%s] [%s] [%s] "
+		"[%s] [%s] [%s] [%s] [%s] [%s] %s [%s] A B op C op D op ... = <outgrid>\n", name, GMT_Rgeo_OPT, GMT_A_OPT, GMT_I_OPT, GMT_V_OPT, GMT_a_OPT, GMT_bi_OPT, GMT_di_OPT,
 		GMT_e_OPT, GMT_f_OPT, GMT_g_OPT, GMT_h_OPT, GMT_i_OPT, GMT_n_OPT, GMT_r_OPT, GMT_s_OPT, GMT_x_OPT, GMT_PAR_OPT);
-	GMT_Message (API, GMT_TIME_NONE, " A B op C op D op ... = <outgrd>\n\n");
 
 	if (level == GMT_SYNOPSIS) return (GMT_MODULE_SYNOPSIS);
 
+	GMT_Usage (API, 1, "<operands>");
```
Comment (meghanrjones) [inline src/grdmath.c:176]: the 'Required Arguments' header and the outgrid argument are missing

File: src/grdmath.c [hunk 2]
```diff
@@ -167,293 +167,296 @@ static void Free_Ctrl (struct GMT_CTRL *GMT, struct GRDMATH_CTRL *C) {	/* Deallo
 static int usage (struct GMTAPI_CTRL *API, int level) {
 	const char *name = gmt_show_name_and_purpose (API, THIS_MODULE_LIB, THIS_MODULE_CLASSIC_NAME, THIS_MODULE_PURPOSE);
 	if (level == GMT_MODULE_PURPOSE) return (GMT_NOERROR);
-	GMT_Message (API, GMT_TIME_NONE, "usage: %s [%s]\n\t[%s]\n\t[-D<resolution>][+f] [%s]\n\t[-M] [-N] [-S] [%s] [%s] [%s] [%s]\n\t[%s] [%s]"
-		" [%s]\n\t[%s] [%s]\n\t[%s] [%s] [%s]\n\t%s [%s]", name, GMT_Rgeo_OPT, GMT_A_OPT, GMT_I_OPT, GMT_V_OPT, GMT_a_OPT, GMT_bi_OPT, GMT_di_OPT,
+	GMT_Usage (API, 0, "usage: %s [%s] [%s] [-D<resolution>[+f] [%s] [-M] [-N] [-S] [%s] [%s] [%s] [%s] [%s] [%s] "
+		"[%s] [%s] [%s] [%s] [%s] [%s] %s [%s] A B op C op D op ... = <outgrid>\n", name, GMT_Rgeo_OPT, GMT_A_OPT, GMT_I_OPT, GMT_V_OPT, GMT_a_OPT, GMT_bi_OPT, GMT_di_OPT,
 		GMT_e_OPT, GMT_f_OPT, GMT_g_OPT, GMT_h_OPT, GMT_i_OPT, GMT_n_OPT, GMT_r_OPT, GMT_s_OPT, GMT_x_OPT, GMT_PAR_OPT);
```
Comment (meghanrjones) [inline src/grdmath.c:172]: -e and -r are listed here but not in rst documentation.

Comment (meghanrjones) [reply src/grdmath.c:172]: -x is not in the usage message despite GMT_x_OPT

Comment (PaulWessel) [reply src/grdmath.c:172]: I am removing -e.  I see -r in the rst docs.  Where do you see it missing?  -x in usage requires OpenMP build so unless you did that you won't see -x listed.

Comment (meghanrjones) [reply src/grdmath.c]: Sorry, another typing mistake. I meant -s, not -r.

Comment (PaulWessel) [reply src/grdmath.c]: Thanks.  I wonder about this one: grdmath has a handful of operators that read tables, hence there is -a, -b, -d, etc.  So I think I hsould change my mind and add back -e and add -e -s to the docs to be consistent.  No reason why tables read by grdmath cannot access some of the table options.

Comment (PaulWessel) [reply src/grdmath.c]: Well, not -s since that is for table output which is not releveant here.  So will remove -s.

**Comment by PaulWessel** [comment]
I think this one is now ready for a re-rewrite by @KristofKoch and/or @meghanrjones. I gave it the same operator wrapping treatment as in gmt math.  I also added missing entries for the output file|stdout and grid.

File: doc/rst/source/grdmath.rst [hunk 3]
```diff
@@ -23,6 +23,7 @@ Synopsis
 [ |SYN_OPT-a| ]
 [ |SYN_OPT-bi| ]
 [ |SYN_OPT-di| ]
+[ |SYN_OPT-e| ]
 [ |SYN_OPT-f| ]
```
Comment (meghanrjones) [inline doc/rst/source/grdmath.rst:27]: is -g needed here in the rst docs? it is included in the usage message.

File: doc/rst/source/grdmath.rst [hunk 4]
```diff
@@ -128,6 +129,9 @@ Optional Arguments
 .. |Add_-di| unicode:: 0x20 .. just an invisible code
 .. include:: explain_-di.rst_
 
+.. |Add_-e| unicode:: 0x20 .. just an invisible code
+.. include:: explain_-e.rst_
+
 .. |Add_-f| unicode:: 0x20 .. just an invisible code
 .. include:: explain_-f.rst_
```
Comment (meghanrjones) [inline doc/rst/source/grdmath.rst:136]: -g may be needed here as well.

**Review by meghanrjones — APPROVED** [review]

File: doc/rst/source/grdmath.rst (hunk 4 above)
Comment (PaulWessel) [reply doc/rst/source/grdmath.rst:137]: Yes, added -g to the docs for consistency.

**Pull request merged**